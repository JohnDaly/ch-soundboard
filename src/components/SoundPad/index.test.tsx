// External Dependencies
import { mount } from 'enzyme'
import enzymeToJSON from 'enzyme-to-json'
import 'jest-styled-components'
import React from 'react'

// Internal Dependencies
import { SoundPad } from './index'

describe('SoundPad', () => {
    it('renders without crashing', () => {
        const wrapper = mount(
            <SoundPad
                title={'Test Sound'}
                trackID={'testSoundID'}
                onClick={jest.fn()}
            />
        )
        expect(wrapper).toBeDefined()
    })

    it('should pass the trackID as an argument in its onClick prop', () => {
        const mockedOnClick = jest.fn()
        const wrapper = mount(
            <SoundPad
                title={'Test Sound'}
                trackID={'testSoundID'}
                onClick={mockedOnClick}
            />
        )

        const padCard = wrapper.find('SoundPadCard')
        expect(padCard).toBeDefined()
        padCard.simulate('click')
        expect(mockedOnClick).toHaveBeenCalled()
        expect(mockedOnClick).toHaveBeenCalledWith('testSoundID')
    })

    it('should match the snapshot', () => {
        const wrapper = mount(
            <SoundPad
                title={'Test Sound'}
                trackID={'testSoundID'}
                onClick={jest.fn()}
            />
        )
        expect(enzymeToJSON(wrapper)).toMatchSnapshot()
    })
})
