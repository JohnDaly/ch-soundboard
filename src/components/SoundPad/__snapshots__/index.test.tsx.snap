// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundPad should match the snapshot 1`] = `
.c0 {
  margin-left: 5px;
  margin-right: 5px;
}

.c1 {
  cursor: pointer;
  width: 40vw;
  height: 40vw;
  max-height: 120px;
  max-width: 120px;
  margin-bottom: 15px;
  border-radius: 3px;
  box-shadow: 2px 3px;
  font-size: 11px;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c1:hover {
  -webkit-filter: brightness(85%);
  filter: brightness(85%);
}

.c1:active {
  box-shadow: 0px 0px black;
  -webkit-filter: brightness(60%);
  filter: brightness(60%);
  -webkit-transform: scale(0.99) translateY(3px) translateX(2px);
  -ms-transform: scale(0.99) translateY(3px) translateX(2px);
  transform: scale(0.99) translateY(3px) translateX(2px);
}

.c2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<SoundPad
  onClick={[Function]}
  title="Test Sound"
  trackID="testSoundID"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
          margin-left: 5px;
          margin-right: 5px;
      ",
          ],
        },
        "displayName": "SoundPad",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    onClick={[Function]}
    title="Test Sound"
    trackID="testSoundID"
  >
    <Component
      className="c0"
      onClick={[Function]}
      title="Test Sound"
      trackID="testSoundID"
    >
      <div
        className="c0"
      >
        <SoundPadCard
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [
                  Object {
                    "className": "card",
                  },
                ],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
                  cursor: pointer;
              
                  /* Dimensions */
                  width: 40vw;
                  height: 40vw;
                  max-height: 120px;
                  max-width: 120px;
              
                  /* Margin/Padding */
                  margin-bottom: 15px;
              
                  /* Border */
                  border-radius: 3px;
                  box-shadow: 2px 3px ",
                    [Function],
                    ";
              
                  /* Font */
                  font-size: 11px;
                  color: ",
                    [Function],
                    ";
              
                  /* Color */
                  background-color: ",
                    [Function],
                    ";
              
                  /* Transition */
                  transition: all 0.2s ease;
              
                  :hover {
                      filter: brightness(85%);
                  }
              
                  :active {
                      box-shadow: 0px 0px black;
                      filter: brightness(60%);
                      transform: scale(0.99) translateY(3px) translateX(2px);
                  }
              ",
                  ],
                },
                "displayName": "SoundPadCard",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="card c1"
              onClick={[Function]}
            >
              <SoundPadBody>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-htpNat",
                        "isStatic": true,
                        "lastClassName": "c2",
                        "rules": Array [
                          "
                        height: 100%;
                        display: flex;
                        align-items: center;
                        justify-content: center;
                    ",
                        ],
                      },
                      "displayName": "SoundPadBody",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-htpNat",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="c2"
                  >
                    Test Sound
                  </div>
                </StyledComponent>
              </SoundPadBody>
            </button>
          </StyledComponent>
        </SoundPadCard>
      </div>
    </Component>
  </StyledComponent>
</SoundPad>
`;
